test/e2e/storage/testsuites/multiattach.go

```diff
@@ -0,0 +1,323 @@
+		// a static volume inside container, not actually creating a new volume per request.
+		if pattern.VolType == testpatterns.PreprovisionedPV {
+			framework.Skipf("This test doesn't work with pre-provisioned volume -- skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		l.resource2.pattern.VolMode = l.resource1.pattern.VolMode
+		testAttachTwoVolumesToOnePod(l)
+	})
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- different volume mode (only <block, filesystem> pattern is tested)
+	//   [volume1]  [volume2]
+	It("should attach two volumes with different volume mode to the same pod", func() {
+		if pattern.VolMode == v1.PersistentVolumeFilesystem {
+			framework.Skipf("Filesystem volume case should be covered by block volume case -- skipping")
+		}
+
+		// Currently, multiple volumes are not generally available for pre-provisoined volume,
+		// because containerized storage servers, such as iSCSI and rbd, are just returning
+		// a static volume inside container, not actually creating a new volume per request.
+		if pattern.VolType == testpatterns.PreprovisionedPV {
+			framework.Skipf("This test doesn't work with pre-provisioned volume -- skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		// Only Block volume test for 1st volume will reach here,
+		// so we should set filesystem for 2nd volume.
+		l.resource2.pattern.VolMode = v1.PersistentVolumeFilesystem
+		testAttachTwoVolumesToOnePod(l)
+	})
+
+	// This tests below configuration:
+	// [pod1] [pod2]
+	// [   node1   ]
+	//   \      /     <- same volume mode
+	//   [volume1]
+	It("should attach the same volume with the same volume mode to the different pod on the same node", func() {
+		init()
+		defer cleanup()
+
+		l.resource2.pattern.VolMode = l.resource1.pattern.VolMode
+		l.resource2.pvc = l.resource1.pvc
+		testAttachOneVolumeToTwoPods(l, true /* sameNode */)
+	})
+
+	// This tests below configuration:
+	//        [pod1] [pod2]
+	// [   node1   ] [   node2   ]
+	//         \      /     <- same volume mode
+	//         [volume1]
+	It("should attach the same volume with the same volume mode to the different pod on different node", func() {
+		init()
+		defer cleanup()
+
+		l.resource2.pattern.VolMode = l.resource1.pattern.VolMode
+		l.resource2.pvc = l.resource1.pvc
+		testAttachOneVolumeToTwoPods(l, false /* sameNode */)
+	})
+}
+
+func getNodesToDeploy(cs clientset.Interface, driverName string, requiredNodeName string, requiresSameNode bool) (string, string) {
+	var node1, node2 string
+	if requiresSameNode {
+		// If node is not specified by requiredNodeName,
+		// decide the node to deploy both 1st pod and 2nd pod
+		// so that they will be deployed on the same node.
+		nodeName := requiredNodeName
+		if nodeName == "" {
+			nodes := framework.GetReadySchedulableNodesOrDie(cs)
+			nodeName = nodes.Items[rand.Intn(len(nodes.Items))].Name
+		}
+		node1 = nodeName
```

Comment:
instead of forcing nodes by setting pod.spec.nodename, I would prefer using PodAffinity/Anti-Affinity so that the pod still goes through the scheduler.  You can probably reuse methods that were implemented in the provisioning suite.